### PR TITLE
Add defaults to the docstrings for algorithms

### PR DESCRIPTION
**Pull request by hanlint**
Fixes: #607 .

* All algorithms that have default arguments are not annotated as, for example:
```
alpha (float, optional): <description>. Default: 0.1
```
* Opportunistically cleans up some missing docstrings and type annotations here and there.

### CODE REVIEW

**Review by dblalock — APPROVED** [review]
Went thru all the pages under under `composer.algorithms` in the generated API reference and changes all LGTM, other than extra backticks in augmix.

I also noticed that public functions/modules in blurpool_layers, factorize_modules, and factorize_core didn't get "Default: whatever" strings, but maybe that was intentional limitation of this PRs scope.

File: composer/algorithms/augmix/augmix.py [hunk 1]
```diff
@@ -56,12 +56,12 @@ def augmix_image(img: ImgT,
...
     Args:
         img (PIL.Image): Image or batch of images to be AugMix'd.
-        severity (int, optional): See :class:`~composer.algorithms.augmix.augmix.AugMix`.
-        depth (int, optional): See :class:`~composer.algorithms.augmix.augmix.AugMix`.
-        width (int, optional): See :class:`~composer.algorithms.augmix.augmix.AugMix`.
-        alpha (float, optional): See :class:`~composer.algorithms.augmix.augmix.AugMix`.
+        severity (int, optional): See :class:`.AugMix``.
+        depth (int, optional): See :class:`.AugMix``.
+        width (int, optional): See :class:`.AugMix``.
+        alpha (float, optional): See :class:`.AugMix``.
```
Comment (hanlint) [inline composer/algorithms/augmix/augmix.py:62]:
```suggestion
        alpha (float, optional): See :class:`.AugMix`.
```

File: composer/algorithms/randaugment/randaugment.py [hunk 2]
```diff
@@ -72,9 +72,9 @@ def _randaugment_pil_image(img: PillowImage, severity: int, depth: int, augmenta
 
 
 class RandAugmentTransform(torch.nn.Module):
-    """Wraps :func:`~composer.algorithms.randaugment.randaugment.randaugment_image` in a
+    """Wraps :func:`.RandAugment_image` in a
```
Comment (growlix) [inline composer/algorithms/randaugment/randaugment.py:75]:
```suggestion
    """Wraps :func:`.randaugment_image` in a
```

File: composer/algorithms/layer_freezing/layer_freezing.py [hunk 3]
```diff
@@ -22,8 +22,8 @@ def freeze_layers(
     model: Model,
     optimizers: Optimizers,
     current_duration: float,
-    freeze_start: float,
-    freeze_level: float,
+    freeze_start: float = 0.5,
+    freeze_level: float = 1.0,
```
Comment (growlix) [inline composer/algorithms/layer_freezing/layer_freezing.py:26]: Does the corresponding `hparams` need to be updated?

**Review by growlix — APPROVED** [review]
Need to change one instance of `RandAugment_image` to `randaugment_image` (see comment) but otherwise lgtm!

File: composer/algorithms/layer_freezing/layer_freezing.py (hunk 3 above)
Comment (hanlint) [reply composer/algorithms/layer_freezing/layer_freezing.py:26]: Nope, they already use these defaults

**Pull request merged**